webrtc-stats.html

```diff
@@ -1216,6 +1216,10 @@
              unsigned long        framesEncoded;
              double               totalEncodeTime;
              double               averageRTCPInterval;
+             RTCQualityLimitationReason  qualityLimitationReason;
+             double                      qualityCpuLimitationDuration;
+             double                      qualityBandwidthLimitationDuration;
+             double                      qualityTotalLimitationDuration;
```

Comment:
...or is record<DOMString, double> qualityLimitationDurations preferred?

Reply:
And should we have enum qualityLimitationMethod = {none, framerate, resolution} (excluding quality due to https://github.com/w3c/webrtc-stats/issues/256#issuecomment-339838513) as per-Harald's suggestion? It sounds useful.

Reply:
I'd like a record<reason, duration> rather than separate counters, because I suspect there will be more types added in the future (GPU restriction?) - a record is more future-proof in this regard than separate counters. I like breaking out the total limitation duration because I suspect that it will be the stat people look at first (during the call: "is it going up or not?")

Reply:
record<> is used in the Fetch IDL in Chrome, so it's implemented and likely to stay around.

Reply:
Done. A side-effect of this is that having record<reason, duration> implicitly means also having the duration for "none" unless we clarify what key values are expected to be implemented for the record. I'm not sure if I should say 'the key is any value of RTCQualityLimitationReason excluding the value "none"', say that "none" is optional to map, or not say anything.

A stats member is optional to implement, but if implemented it should be spec-compliant. What about record<> types? I expect you either implement none of it or all of it, so we should be clear about whether "none" needs to be implemented I suppose? PTAL and comment about this.